src/views/comparison/ChartB.js

```diff
@@ -179,7 +179,236 @@ export class ChartB {
       this.topData.getAllFeatures().catch((err) => []),
       this.botData.getAllFeatures().catch((err) => []),
     ]).then(([topFeatures, botFeatures]) => {
-      //get 5-statistics for interstitial area
+      //function to draw violin plot for one experiments
+      function violin(
```

Comment:
I think you could further simplify this function to draw one violin at a time probably and then call it 6 times since there is still some code repeition